src/database/models/categories.rs

```diff
@@ -303,11 +304,14 @@ impl Loader {
         let result = sqlx::query!(
             "
             SELECT l.id id, l.loader loader, l.icon icon,
+            ARRAY_AGG(DISTINCT lrt.loader) supported_loaders,
             ARRAY_AGG(DISTINCT pt.name) project_types
             FROM loaders l
-            LEFT OUTER JOIN loaders_project_types lpt ON joining_loader_id = l.id
-            LEFT OUTER JOIN project_types pt ON lpt.joining_project_type_id = pt.id
-            GROUP BY l.id;
+                     LEFT OUTER JOIN loaders_project_types lpt ON joining_loader_id = l.id
+                     LEFT OUTER JOIN project_types pt ON lpt.joining_project_type_id = pt.id
+                     LEFT OUTER JOIN loaders_relations lr on l.id = lr.child_loader
+                     LEFT OUTER JOIN loaders lrt on lr.parent_loader = lrt.id
+            GROUP BY l.id
```

Comment:
Avoid LEFT JOINs (they have a huge peformance cost). You can look at other examples in the codebase where they were replaced for multiple queries.

Reply:
The performance impact in this specific case is really low (the number of records won't be really high), and in order to make this performant, caching may be needed for that.